### PR TITLE
Make optimizer an abstract class

### LINKED ISSUE #68: @abstractmethod in Optimizer does not work
We have to set the meta class of Optimizer to ABCMeta like this is done for the ContextualOptimizer.

### PR DESCRIPTION
**Pull request by AlexanderFabisch**
Fixes #68

### CODE REVIEW

**Comment by AlexanderFabisch** [comment]
@ChristophPetzoldt Could you also review this pull request?

File: bolero/optimizer/test/test_optimizer.py [hunk 1]
```diff
@@ -2,14 +2,27 @@
...
 from bolero.optimizer import Optimizer, ContextualOptimizer
 from bolero.utils.testing import assert_pickle
-from nose.tools import assert_false, assert_true, assert_equal
+from nose.tools import (assert_false, assert_true, assert_equal,
+                        assert_raises_regexp)
 from numpy.testing import assert_array_equal
 
 
 ALL_OPTIMIZERS = all_subclasses(Optimizer)
 ALL_CONTEXTUALOPTIMIZERS = all_subclasses(ContextualOptimizer)
```
Comment (ChristophPetzoldt) [inline bolero/optimizer/test/test_optimizer.py]: all_subclasses returns an empty list.. So some tests actually do not run.

File: bolero/optimizer/optimizer.py [hunk 2]
```diff
@@ -91,6 +91,7 @@ def best_policy(self):
 
 class Optimizer(Base):
     """Common interface for (non-contextual) optimizers."""
+    __metaclass__ = ABCMeta
```
Comment (ChristophPetzoldt) [inline bolero/optimizer/optimizer.py]: Looks good to me.

File: bolero/optimizer/test/test_optimizer.py (hunk 1 above)
Comment (AlexanderFabisch) [reply bolero/optimizer/test/test_optimizer.py]: Uh, didn't see that before. I introduced this bug here: https://github.com/rock-learning/bolero/commit/abd258d2e61fc635b54ac4eec5848308f8acf11b#diff-39d53a7dd0adc4b9b1e549b967aa6034

Comment (ChristophPetzoldt) [reply bolero/optimizer/test/test_optimizer.py]: Oops, unfortunate commit, but easy to fix then, if its just this code indentation ;)

**Comment by AlexanderFabisch** [comment]
Merged to master, but now we have a problem with CircleCI. I'm on it...

**Pull request closed without merging**